src/app/app.component.spec.ts

```diff
@@ -14,4 +14,123 @@ describe('AppComponent', () => {
       ],
     }).compileComponents();
   });
+
+  const TRIPLE_BANANA_SHARE_URL = "https://triplebanana.dev/share"
+
+  const USER_AGENT_ANDROID_CHROME_BROWSER =
+    "Mozilla/5.0 (Linux; Android 12; SM-G996N) \
+    AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Mobile Safari/537.36"
+  const USER_AGENT_ANDROID_KAKAOTALK_WEB_VIEW =
+    "Mozilla/5.0 (Android; Mobile; rv:13.0) \
+    Gecko/13.0 Firefox/13.0 KAKAOTALK"
+
+  const USER_AGENT_LINUX_CHROME =
+    "Mozilla/5.0 (X11; Linux x86_64) \
+    AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.61 Safari/537.36"
+
+  it('should move android intent including target url when use chrome browser', fakeAsync(() => {
+    // given
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+
+    const shareUrl = "https://google.com"
+    spyOn<any>(app, 'getCurrentUrl').and.returnValue(generateShareLink(shareUrl))
+    const moveToUrl = spyOn<any>(app, 'moveToUrl')
+    spyOnProperty(navigator, 'userAgent').and.returnValue(USER_AGENT_ANDROID_CHROME_BROWSER)
+
+    // when
+    fixture.detectChanges()
+    tick()
+
+    // then
+    const calledIntent = new URL(moveToUrl.calls.mostRecent().args[0] as string)
+    expect(calledIntent.protocol).toEqual("intent:")
+    expect(calledIntent.pathname.includes("google.com")).toBeTruthy()
```

Comment:
Can we use `toEqual` for the `calledIntent.pathname` instead of `includes` and `toBeTruthy`?

Reply:
I've checked this, seems URL class does not support "intent:" scheme as my purpose
the full of intent url is 

intent://google.com/#Intent;scheme=http;package=org.triple.banana;end

when parse it as URL,

> hash: "#Intent;scheme=http;package=org.triple.banana;end"
host: ""
hostname: ""
href: "intent://google.com/#Intent;scheme=http;package=org.triple.banana;end"
origin: "null"
password: ""
pathname: "//google.com/"
port: ""
protocol: "intent:"

while new URL("https://google.com")

> hash: ""
host: "google.com"
hostname: "google.com"
href: "https://google.com/"
origin: "https://google.com"
password: ""
pathname: "/"
port: ""
protocol: "https:"
search: ""
searchParams: URLSearchParams {}
username: ""

so it seems using pathname here may not be a good idea.